[Bug]: Flink failing to insert after restart of app

### Summary

App was closed with connection error with the DB
App service was restarted but flink job did not successfully resumed
Specifically the sinking step(IcebergSink iceberg.dl.messages: Writer) failed

### Steps to Reproduce

_No response_

### Component

App, Flink

### Relevant Logs

App

```shell
plutus_app  | Attempt 1 at connecting failed: TimeoutError:
plutus_app  | Attempt 2 at connecting failed: TimeoutError:
plutus_app  | Attempt 3 at connecting failed: TimeoutError:
plutus_app  | Attempt 4 at connecting failed: TimeoutError:
plutus_app  | Attempt 5 at connecting failed: TimeoutError:
plutus_app  | Attempt 6 at connecting failed: TimeoutError:
plutus_app  | Attempt 1 at connecting failed: TimeoutError:
plutus_app  | Attempt 2 at connecting failed: TimeoutError:
plutus_app  | Attempt 3 at connecting failed: TimeoutError:
plutus_app  | Attempt 4 at connecting failed: TimeoutError:
plutus_app  | Attempt 5 at connecting failed: TimeoutError:
plutus_app  | Attempt 6 at connecting failed: TimeoutError:
plutus_app  | Attempt 1 at connecting failed: TimeoutError:
plutus_app  | Attempt 2 at connecting failed: TimeoutError:
plutus_app  | Attempt 3 at connecting failed: TimeoutError:
plutus_app  | Attempt 4 at connecting failed: TimeoutError:
plutus_app  | Attempt 5 at connecting failed: TimeoutError:
plutus_app  | Attempt 6 at connecting failed: TimeoutError:
plutus_app  | Attempt 1 at connecting failed: TimeoutError:
plutus_app  | Attempt 2 at connecting failed: TimeoutError:
plutus_app  | Attempt 3 at connecting failed: TimeoutError:
plutus_app  | Attempt 4 at connecting failed: TimeoutError:
plutus_app  | Attempt 5 at connecting failed: TimeoutError:
plutus_app  | Attempt 6 at connecting failed: TimeoutError:
plutus_app  | Attempt 1 at connecting failed: TimeoutError:
plutus_app  | Attempt 2 at connecting failed: TimeoutError:
plutus_app  | Attempt 3 at connecting failed: TimeoutError:
plutus_app  | Attempt 4 at connecting failed: TimeoutError:
plutus_app  | Attempt 5 at connecting failed: TimeoutError:
plutus_app  | Attempt 6 at connecting failed: TimeoutError:
plutus_app  | Automatic reconnection failed 5 time(s)
plutus_app  | Future exception was never retrieved
plutus_app  | future: <Future finished exception=ConnectionError('Connection to Telegram failed 5 time(s)')>
plutus_app  | ConnectionError: Connection to Telegram failed 5 time(s)
plutus_app  | Traceback (most recent call last):
plutus_app  |   File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
plutus_app  |     return _run_code(code, main_globals, None,
plutus_app  |   File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
plutus_app  |     exec(code, run_globals)
plutus_app  |   File "/app/plutus/main.py", line 28, in <module>
plutus_app  |     client.run_until_disconnected()
plutus_app  |   File "/usr/local/lib/python3.10/site-packages/telethon/client/updates.py", line 96, in run_until_disconnected
plutus_app  |     return self.loop.run_until_complete(self._run_until_disconnected())
plutus_app  |   File "/usr/local/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
plutus_app  |     return future.result()
plutus_app  |   File "/usr/local/lib/python3.10/site-packages/telethon/client/updates.py", line 36, in _run_until_disconnected
plutus_app  |     result = await self.disconnected
plutus_app  |   File "/usr/local/lib/python3.10/site-packages/telethon/client/updates.py", line 354, in _update_loop
plutus_app  |     diff = await self(get_diff)
plutus_app  |   File "/usr/local/lib/python3.10/site-packages/telethon/client/users.py", line 30, in __call__
plutus_app  |     return await self._call(self._sender, request, ordered=ordered)
plutus_app  |   File "/usr/local/lib/python3.10/site-packages/telethon/client/users.py", line 92, in _call
plutus_app  |     result = await future
plutus_app  | ConnectionError: Connection to Telegram failed 5 time(s)
plutus_app exited with code 1 (restarting)
plutus_app  | Loaded config for: []
plutus_app  | Database initialized.
plutus_app  | Client is starting...
plutus_app  | Attempt 1 at connecting failed: TimeoutError:
plutus_app  | Attempt 2 at connecting failed: TimeoutError:
plutus_app  | Attempt 3 at connecting failed: TimeoutError:
plutus_app  | Attempt 4 at connecting failed: TimeoutError:
plutus_app  | Attempt 5 at connecting failed: TimeoutError:
plutus_app  | Attempt 6 at connecting failed: TimeoutError:
plutus_app  | Traceback (most recent call last):
plutus_app  |   File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
plutus_app  |     return _run_code(code, main_globals, None,
plutus_app  |   File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
plutus_app  |     exec(code, run_globals)
plutus_app  |   File "/app/plutus/main.py", line 27, in <module>
plutus_app  |     client.start()
plutus_app  |   File "/usr/local/lib/python3.10/site-packages/telethon/client/auth.py", line 128, in start
plutus_app  |     else self.loop.run_until_complete(coro)
plutus_app  |   File "/usr/local/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
plutus_app  |     return future.result()
plutus_app  |   File "/usr/local/lib/python3.10/site-packages/telethon/client/auth.py", line 135, in _start
plutus_app  |     await self.connect()
plutus_app  |   File "/usr/local/lib/python3.10/site-packages/telethon/client/telegrambaseclient.py", line 552, in connect
plutus_app  |     if not await self._sender.connect(self._connection(
plutus_app  |   File "/usr/local/lib/python3.10/site-packages/telethon/network/mtprotosender.py", line 133, in connect
plutus_app  |     await self._connect()
plutus_app  |   File "/usr/local/lib/python3.10/site-packages/telethon/network/mtprotosender.py", line 266, in _connect
plutus_app  |     raise ConnectionError('Connection to Telegram failed {} time(s)'.format(self._retries))
plutus_app  | ConnectionError: Connection to Telegram failed 5 time(s)
plutus_app exited with code 1 (restarting)
```


Flink

```
2026-02-12 08:14:32
org.apache.iceberg.exceptions.RuntimeIOException: Failed to create file: s3a://data-bucket/warehouse/dl/messages/metadata/34eef572-ba9c-4cf6-9956-cffe53307cfa-m0.avro
	at org.apache.iceberg.hadoop.HadoopOutputFile.createOrOverwrite(HadoopOutputFile.java:87)
	at org.apache.iceberg.avro.AvroFileAppender.<init>(AvroFileAppender.java:56)
	at org.apache.iceberg.avro.Avro$WriteBuilder.build(Avro.java:211)
	at org.apache.iceberg.ManifestWriter$V2Writer.newAppender(ManifestWriter.java:401)
	at org.apache.iceberg.ManifestWriter.<init>(ManifestWriter.java:62)
	at org.apache.iceberg.ManifestWriter$V2Writer.<init>(ManifestWriter.java:378)
	at org.apache.iceberg.ManifestFiles.newWriter(ManifestFiles.java:208)
	at org.apache.iceberg.ManifestFiles.write(ManifestFiles.java:185)
	at org.apache.iceberg.SnapshotProducer.newManifestWriter(SnapshotProducer.java:570)
	at org.apache.iceberg.SnapshotProducer.lambda$newRollingManifestWriter$3(SnapshotProducer.java:580)
	at org.apache.iceberg.RollingManifestWriter.currentWriter(RollingManifestWriter.java:111)
	at org.apache.iceberg.RollingManifestWriter.add(RollingManifestWriter.java:58)
	at java.base/java.lang.Iterable.forEach(Unknown Source)
	at org.apache.iceberg.SnapshotProducer.writeDataFileGroup(SnapshotProducer.java:632)
	at org.apache.iceberg.SnapshotProducer.lambda$writeDataManifests$5(SnapshotProducer.java:621)
	at org.apache.iceberg.SnapshotProducer.lambda$writeManifests$8(SnapshotProducer.java:689)
	at org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:413)
	at org.apache.iceberg.util.Tasks$Builder$1.run(Tasks.java:315)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.apache.hadoop.fs.s3a.AWSClientIOException: getFileStatus on s3a://data-bucket/warehouse/dl/messages/metadata/34eef572-ba9c-4cf6-9956-cffe53307cfa-m0.avro: com.amazonaws.SdkClientException: Unable to unmarshall response (Trying to access closed classloader. Please check if you store classloaders directly or indirectly in static fields. If the stacktrace suggests that the leak occurs in a third party library and cannot be fixed immediately, you can disable this check with the configuration 'classloader.check-leaked-classloader'.). Response Code: 200, Response Text: OK: Unable to unmarshall response (Trying to access closed classloader. Please check if you store classloaders directly or indirectly in static fields. If the stacktrace suggests that the leak occurs in a third party library and cannot be fixed immediately, you can disable this check with the configuration 'classloader.check-leaked-classloader'.). Response Code: 200, Response Text: OK
	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:214)
	at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:175)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.s3GetFileStatus(S3AFileSystem.java:3861)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.innerGetFileStatus(S3AFileSystem.java:3688)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.innerCreateFile(S3AFileSystem.java:1682)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.lambda$create$6(S3AFileSystem.java:1646)
	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.lambda$trackDurationOfOperation$5(IOStatisticsBinding.java:499)
	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.trackDuration(IOStatisticsBinding.java:444)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.trackDurationAndSpan(S3AFileSystem.java:2337)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.trackDurationAndSpan(S3AFileSystem.java:2356)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.create(S3AFileSystem.java:1645)
	at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:1195)
	at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:1175)
	at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:1064)
	at org.apache.iceberg.hadoop.HadoopOutputFile.createOrOverwrite(HadoopOutputFile.java:85)
	... 22 more
Caused by: com.amazonaws.SdkClientException: Unable to unmarshall response (Trying to access closed classloader. Please check if you store classloaders directly or indirectly in static fields. If the stacktrace suggests that the leak occurs in a third party library and cannot be fixed immediately, you can disable this check with the configuration 'classloader.check-leaked-classloader'.). Response Code: 200, Response Text: OK
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleResponse(AmazonHttpClient.java:1818)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleSuccessResponse(AmazonHttpClient.java:1477)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1384)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1157)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:814)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:781)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:755)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:715)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:697)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:561)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:541)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5456)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5403)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5397)
	at com.amazonaws.services.s3.AmazonS3Client.listObjectsV2(AmazonS3Client.java:971)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.lambda$listObjects$11(S3AFileSystem.java:2595)
	at org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding.lambda$trackDurationOfOperation$5(IOStatisticsBinding.java:499)
	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:414)
	at org.apache.hadoop.fs.s3a.Invoker.retryUntranslated(Invoker.java:377)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.listObjects(S3AFileSystem.java:2586)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.s3GetFileStatus(S3AFileSystem.java:3832)
	... 34 more
Caused by: java.lang.IllegalStateException: Trying to access closed classloader. Please check if you store classloaders directly or indirectly in static fields. If the stacktrace suggests that the leak occurs in a third party library and cannot be fixed immediately, you can disable this check with the configuration 'classloader.check-leaked-classloader'.
	at org.apache.flink.util.FlinkUserCodeClassLoaders$SafetyNetWrapperClassLoader.ensureInner(FlinkUserCodeClassLoaders.java:189)
	at org.apache.flink.util.FlinkUserCodeClassLoaders$SafetyNetWrapperClassLoader.getResources(FlinkUserCodeClassLoaders.java:224)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(Unknown Source)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(Unknown Source)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(Unknown Source)
	at java.base/java.util.ServiceLoader$2.hasNext(Unknown Source)
	at java.base/java.util.ServiceLoader$3.hasNext(Unknown Source)
	at java.xml/org.xml.sax.helpers.XMLReaderFactory.lambda$findServiceProvider$0(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at java.xml/org.xml.sax.helpers.XMLReaderFactory.findServiceProvider(Unknown Source)
	at java.xml/org.xml.sax.helpers.XMLReaderFactory.createXMLReader(Unknown Source)
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.<init>(XmlResponsesSaxParser.java:135)
	at com.amazonaws.services.s3.model.transform.Unmarshallers$ListObjectsV2Unmarshaller.unmarshall(Unmarshallers.java:135)
	at com.amazonaws.services.s3.model.transform.Unmarshallers$ListObjectsV2Unmarshaller.unmarshall(Unmarshallers.java:125)
	at com.amazonaws.services.s3.internal.S3XmlResponseHandler.handle(S3XmlResponseHandler.java:62)
	at com.amazonaws.services.s3.internal.S3XmlResponseHandler.handle(S3XmlResponseHandler.java:31)
	at com.amazonaws.http.response.AwsResponseHandlerAdapter.handle(AwsResponseHandlerAdapter.java:69)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleResponse(AmazonHttpClient.java:1794)
	... 54 more
```

--- a/docker-compose.flink.yml
+++ b/docker-compose.flink.yml
@@ -10,6 +10,7 @@ x-flink-common:
       metrics.reporter.prom.factory.class: org.apache.flink.metrics.prometheus.PrometheusReporterFactory
       metrics.reporter.prom.port: 9091
       fs.allowed-fallback-filesystems: s3;s3a
+      classloader.resolve-order: parent-first
   networks:
     - data-net
 

--- a/docker-compose.flink.yml
+++ b/docker-compose.flink.yml
@@ -10,7 +10,7 @@ x-flink-common:
       metrics.reporter.prom.factory.class: org.apache.flink.metrics.prometheus.PrometheusReporterFactory
       metrics.reporter.prom.port: 9091
       fs.allowed-fallback-filesystems: s3;s3a
-      classloader.resolve-order: parent-first
+      classloader.parent-first-patterns.additional: org.apache.hadoop;com.amazonaws;org.apache.http
   networks:
     - data-net
 

